### PR TITLE
catalog: group, rename, and move filter components to catalog-react

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

Ended up feeling that this was another change we wanted in for 1.0, although we'll leave it as a deprecation for now.

It switches this:

```tsx
<FilteredEntityLayout>
  <FilterContainer>
    {/* filter drawer, for example <EntityTypePicker /> and friends */}
  </FilterContainer>
  <EntityListContainer>
    {/* content view, for example a <CatalogTable /> */}
  </EntityListContainer>
</FilteredEntityLayout>
```

to

```tsx
<CatalogFilterLayout>
  <CatalogFilterLayout.Filters>
    {/* filter drawer, for example <EntityTypePicker /> and friends */}
  </CatalogFilterLayout.Filters>
  <CatalogFilterLayout.Content>
    {/* content view, for example a <CatalogTable /> */}
  </CatalogFilterLayout.Content>
</CatalogFilterLayout>
```

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/nasty-chefs-dream.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+'@backstage/plugin-catalog-react': patch
+---
+
+Added `CatalogFilterLayout`, which is replaces `FilteredEntityLayout` from `@backstage/plugin-catalog`, as well as `FilterContainer` and `EntityListContainer`. It is used like this:
```
Comment (freben) [inline .changeset/nasty-chefs-dream.md:5]:
```suggestion
Added `CatalogFilterLayout`, which replaces `FilteredEntityLayout` from `@backstage/plugin-catalog`, as well as `FilterContainer` and `EntityListContainer`. It is used like this:
```

Signed-off-by: Fredrik Adelöw <freben@gmail.com>

File: docs/features/software-catalog/catalog-customization.md [hunk 2]
```diff
@@ -151,18 +151,18 @@ export const CustomCatalogPage = ({
...
     ...
         <EntityListProvider>
-          <FilteredEntityLayout>
-            <FilterContainer>
+          <CatalogFilterLayout>
+            <CatalogFilterLayout.Filters>
               <EntityKindPicker initialFilter="component" hidden />
               <EntityTypePicker />
               <UserListPicker initialFilter={initiallySelectedFilter} />
 +             <EntitySecurityTierPicker />
```
Comment (freben) [inline docs/features/software-catalog/catalog-customization.md:159]: spurious addition :)

File: plugins/catalog/api-report.md [hunk 3]
```diff
@@ -319,15 +319,13 @@ export interface EntitySwitchProps {
...
+// @public @deprecated (undocumented)
+export const FilterContainer: (props: { children: ReactNode }) => JSX.Element;
 
-// @public (undocumented)
-export function FilteredEntityLayout(props: {
-  children: React_2.ReactNode;
-}): JSX.Element;
+// @public @deprecated (undocumented)
+export const FilteredEntityLayout: (props: {
+  children?: React.ReactNode;
```
Comment (freben) [inline plugins/catalog/api-report.md:327]: resulted in a new question mark on this - meh, that's ok

**Review by freben — APPROVED** [review]

File: docs/features/software-catalog/catalog-customization.md (hunk 2 above)
Comment (Rugvip) [reply docs/features/software-catalog/catalog-customization.md:159]: It's just the diff syntax being highlighted 😅

File: plugins/catalog/api-report.md (hunk 3 above)
Comment (Rugvip) [reply plugins/catalog/api-report.md:327]: 👍, nah that should totally be fixed, was doing the cast wrong

File: docs/features/software-catalog/catalog-customization.md (hunk 2 above)
Comment (freben) [reply docs/features/software-catalog/catalog-customization.md:159]: ah man

**Pull request merged**